src/python/gudhi/wasserstein/wasserstein.py

```diff
@@ -85,6 +99,9 @@ def wasserstein_distance(X, Y, matching=False, order=2., internal_p=2.):
     :param order: exponent for Wasserstein; Default value is 2.
     :param internal_p: Ground metric on the (upper-half) plane (i.e. norm L^p in R^2);
                        Default value is 2 (Euclidean norm).
+    :param enable_autodiff: If X and Y are torch.tensor, tensorflow.Tensor or jax.numpy.ndarray, make the computation
+        transparent to automatic differentiation.
```

Comment:
> you cannot "create" new points

Better not, or the gradient (direction which increases the function) might tell me to pull an infinite number of points from the diagonal.

> nor merge a point into the diagonal

When you compute the gradient, you only look at what happens in an arbitrarily small neighborhood, there is no such thing as reaching some position over there. But following the (opposite of the) gradient can push a point in the direction of the diagonal, and in a gradient descent, it can end up on the diagonal.